packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/SchemaMapper.java

```diff
@@ -49,10 +49,10 @@ public IType map(Schema value) {
     private IType createSchemaType(Schema value) {
         if (value instanceof PrimitiveSchema) {
             return Mappers.getPrimitiveMapper().map((PrimitiveSchema) value);
-        } else if (value instanceof ChoiceSchema) {
-            return Mappers.getChoiceMapper().map((ChoiceSchema) value);
         } else if (value instanceof SealedChoiceSchema) {
             return Mappers.getSealedChoiceMapper().map((SealedChoiceSchema) value);
+        } else if (value instanceof ChoiceSchema) {
```

Comment:
Why did we change the order of evaluation here?

Reply:
It's a previous codegen bug: https://github.com/microsoft/typespec/pull/4492#discussion_r1779981094

Here the bug appears in modelProperty. For model itself, we don't have this bug since it's handled elsewhere: 
https://github.com/microsoft/typespec/blob/3804d7a5881d86c8e064a41f7eb6b11a070c92eb/packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/ClientMapper.java#L94-L104